persistent/Database/Persist/Quasi.hs

```diff
@@ -541,31 +539,46 @@ parse :: PersistSettings -> Text -> [EntityDef]
 parse ps = maybe [] (parseLines ps) . preparse
 
 preparse :: Text -> Maybe (NonEmpty Line)
-preparse =
-    NEL.nonEmpty
-        . removeSpaces
-        . filter (not . empty)
-        . map tokenize
-        . T.lines
+preparse txt = do
+    lns <- NEL.nonEmpty (T.lines txt)
+    NEL.nonEmpty $ mapMaybe parseLine (NEL.toList lns)
+
+-- TODO: refactor to return (Line' NonEmpty)
```

Comment:
I noticed I'd left this in, but there is no real explanation here... This is something I wanted to change, but was worried that it may inflate this initial PR, I will address this following this PR I think. Effectively we can parse a `Line' NonEmpty` upfront here, which means we can greatly simplify the implementation of `Line'` as it no longer needs to be polymorphic as the the type used for the 'collection' of tokens. I will do this as another change though after this one.

If you don't want `TODOs` lying around I can remove this :+1:

Reply:
Could you leave this PR comment in the note? That'd be great for whoever sees this TODO later on. THanks!